test/test_supervisor.rb

```diff
@@ -161,7 +161,7 @@ def test_rpc_server
 
     assert{ $log.out.logs.first.end_with?(info_msg) }
   ensure
-    $log.out.reset
+    $log.out.reset unless $log.out.is_a?(Fluent::LogDeviceIO)
```

Comment:
This method(`reset`) is only implemented in  Fluent::Test::DummyLogDevice.  I think the following is better .

```suggestion
    $log.out.reset if $log.out.is_a?(Fluent::Test::DummyLogDevice)
```